joomla/joomla-cms
# [4.0]Improve component dispatcher

Pull Request for Issue # .

### Summary of Changes
This PR makes some small improvements to current component dispatcher implement:

1. Introduce option property so that component dispatcher is not coupled with $this->app->scope anymore

2. Use standard namespace for component if $namespace is empty in specific component dispatcher class. Honestly, I don't know if we want to allow magic calculation like this, so looking for feedback.

### Testing Instructions
1. Install latest Joomla 4.0-dev branch
2. Apply patch
3. Login to administrator area of the site, just access to the two namespaced components com_content, com_banners, make sure it is loading as expected (no need for further testings, just make sure banners list and articles list is being displayed)
